### PR TITLE
Welford layernorm perf improvements + expanded layernorm/RMSnorm test coverage

### PR DESCRIPTION
**Pull request by rmillerTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/30680

### Problem description
This PR implements a few performance enhancements to the Welford layernorm paths (interleaved and sharded):

1. Pre-compute reciprocals for Welford LLK in program factory and shard to cores
2. Remove the generation and cb_wait for scaler CB's that are not used in Welford path
3. Get rid of a binary_op_init that was unnecessary

The overall improvement in e2e time for the DRAM interleaved path over the previous implementation (computing the reciprocals in the Welford LLK using division) is shown in the table below:

Baseline: Legacy layernorm implementation (using reduce_tile)
Before: Welford layernorm using division
After: Welford layernorm using pre-computed reciprocals

E2E op times averaged over 10 runs (DRAM interleaved):
image

This PR also greatly expands layernorm and rms norm test coverage to include sharded rms norm and row-major gamma/beta to invoke the row-major reader kernels.

### What's changed
Described above

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes https://github.com/tenstorrent/tt-metal/actions/runs/18921802999
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/18921808825
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/18942142377
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by bbradelTT** [comment]
Given the description, the title should be changed to "Welford layernorm perf improvements with row-major reader paths fix" and the description updated to include a quick summary of the perf improvements.
Also, please remove the second "Problem description".

File: tests/ttnn/unit_tests/operations/fused/test_layer_norm_sharded.py [hunk 1]
```diff
@@ -6,9 +6,8 @@
 import torch
 import ttnn
 
-from tests.ttnn.utils_for_testing import assert_with_pcc
+import tests.ttnn.unit_tests.operations.fused.sharded_test_utils as stu
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/fused/test_layer_norm_sharded.py]: Please do from ... import ...
stu is not a well known acronym and sharded_test_utils is too long.

File: ttnn/cpp/ttnn/operations/normalization/layernorm/device/kernels/dataflow/writer_unary_sharded_ln_rm_gb.cpp [hunk 2]
```diff
@@ -46,19 +46,21 @@ void kernel_main() {
 
     const uint32_t out_single_tile_size_bytes = get_tile_size(cb_out);
 
-    {
+    if constexpr (!use_welford) {
         constexpr uint32_t cb_in_2 = tt::CBIndex::c_2;
         const uint32_t scalar_w = get_arg_val<uint32_t>(1);
         generate_reduce_scaler(cb_in_2, scalar_w);
+
+        constexpr uint32_t eps_cb_id = 3;
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/layernorm/device/kernels/dataflow/writer_unary_sharded_ln_rm_gb.cpp]: Would `tt::CBIndex::c_3` be better?

**Review by bbradelTT — APPROVED** [review]

**Comment by rmillerTT** [comment]
Model regression had one failure from a down node, and T3K demo got the sign off from  models team. L2 nightly passed https://github.com/tenstorrent/tt-metal/actions/runs/18947795937/job/54121661581

**Review by mtairum — APPROVED** [review]
Confirmed in the T3K demo pipelines that the accuracy is good, and no new issues were introduced there.
